db/db.go

```diff
@@ -125,24 +126,39 @@ func (d DB) NewAuthor(a Author) error {
 	return nil
 }
 
-// func (d DB) ReadAuthors(a int) (bool, error) {
-// 	// check if author name (Author.Name) is in DB
-// 	// input author name as string, look up all rows of Author.ID in Post struct, get list of float32s and query
-// 	var authorChecked int
+func (d DB) AuthorExists(a string) (bool, error) {
+	var (
+		author  string
+		authors []string
+	)
+
+	stmt, err := d.client.Prepare("SELECT name FROM authors")
+	if err != nil {
+		return false, fmt.Errorf("failed to select name: %w", err)
+	}
+	defer stmt.Close()
+
+	rows, err := stmt.Query()
+	if err != nil {
+		return false, fmt.Errorf("failed to query: %w", err)
+	}
+	defer rows.Close()
 
-// 	stmt, err := d.client.Prepare("SELECT AuthorID from posts")
-// 	if err != nil {
-// 		return false, fmt.Errorf("failed to select AuthorID: %w", err)
-// 	}
-// 	defer stmt.Close()
+	for rows.Next() {
+		err = rows.Scan(&author)
+		if err != nil {
+			return false, fmt.Errorf("failed to scan %w", err)
+		}
 
-// 	err = stmt.QueryRow(a).Scan(&authorChecked)
-// 	if err != nil {
-// 		return false, fmt.Errorf("failed to queryRow: %w", err)
-// 	}
+		authors = append(authors, author)
+	}
+
+	if slices.Contains(authors, a) {
+		return true, nil
+	}
```

Comment:
This whole thing could be changed to a single SQL query that would be way more efficient than fetching _all_ authors.

Imagine there are 1 million rows, that would be very inefficient to query all, parse them into users, load them into an list, and check if the list contains the author's name.